coffe/hardblock_functions.py

```diff
@@ -156,10 +157,173 @@ def run_synth(flow_settings,clock_period,wire_selection):
+    "globalNetConnect " + flow_settings["pwr_pin"] + " -type tiehi -inst *",
+    "globalNetConnect " + flow_settings["gnd_pin"] + " -type tielo -inst *",
+    "sroute -connect { blockPin padPin padRing corePin floatingStripe } -layerChangeRange { "+ " ".join([metal_layer_bottom,metal_layer_top]) + " }"\
+     + " -blockPinTarget { nearestRingStripe nearestTarget } -padPinPortConnect { allPort oneGeom } -checkAlignedSecondaryPin 1 -blockPin useLef -allowJogging 1"\
+     + " -crossoverViaBottomLayer " + metal_layer_bottom + " -targetViaBottomLayer " + metal_layer_bottom + " -allowLayerChange 1"\
+     + " -targetViaTopLayer " + metal_layer_top + " -crossoverViaTopLayer " + metal_layer_top + " -nets {" + " ".join([flow_settings["gnd_net"],flow_settings["pwr_net"]]) +  "}",
+    "routeDesign -globalDetail",
+    "setExtractRCMode -engine postRoute",
+    "extractRC",
+    "buildTimingGraph",
+    "setAnalysisMode -analysisType onChipVariation -cppr both",
+    "timeDesign -postRoute -outDir " +  os.path.join(flow_settings["pr_folder"],"timeDesignReports"),
+    "optDesign -postRoute -outDir " +  os.path.join(flow_settings["pr_folder"],"optDesignReports"),
+    "verify_drc -report " + os.path.join(flow_settings["pr_folder"],"geom.rpt"),
+    "verifyConnectivity -type all -report " + os.path.join(flow_settings["pr_folder"],"conn.rpt"),
+    "report_timing > " + os.path.join(flow_settings["pr_folder"],flow_settings["top_level"] +"setup_timing.rpt"),
+    "setAnalysisMode -checkType hold",
+    "report_timing > " + os.path.join(flow_settings["pr_folder"],"hold_timing.rpt"),
+    "report_power > " + os.path.join(flow_settings["pr_folder"],"power.rpt"),
+    "report_constraint -all_violators > " + os.path.join(flow_settings["pr_folder"],"violators.rpt"),
+    "summaryReport -outFile " + os.path.join(flow_settings["pr_folder"],"pr_report.txt"),
+    "saveNetlist " + os.path.join(flow_settings["pr_folder"],"netlist.v"),
+    "saveDesign " +  os.path.join(flow_settings["pr_folder"],"design.enc"),
+    "rcOut -spef " +  os.path.join(flow_settings["pr_folder"],"spef.spef"),
+    "write_sdf -ideal_clock_network " + os.path.join(flow_settings["pr_folder"],"sdf.sdf"),
+    "exit"]
+  #If the user specified a layer mapping file, then use that. Otherwise, just let the tool create a default one.
+  if flow_settings['map_file'] != "None":
+    file_lines.append("streamOut " +  os.path.join(flow_settings["pr_folder"],"final.gds2") + " -mapFile " + flow_settings["map_file"] + " -stripes 1 -units 1000 -mode ALL")
+  else:
+    file_lines.append("streamOut " +  os.path.join(flow_settings["pr_folder"],"final.gds2") + " -stripes 1 -units 1000 -mode ALL")
+  for line in file_lines:
+    file_write_ln(file,line)
+  file.close()
+  return fname
+
+def write_innovus_init_script(flow_settings,view_fname):
+  flow_settings["lef_files"] = flow_settings["lef_files"].strip("\"")
```

Comment:
Write a docstring saying what this function does. Looks like it is also specific to our CMC setup; that's OK but it should be mentioned.